tests/unit/test_adjust.py

```diff
@@ -0,0 +1,120 @@
+
+    # using logical operators
+    component = firefox && arch = ppc64: False
+
+    arch = s390x: PHASES=novalgrind # modify environment
+
+    # try special operators
+    collection contains httpd24 && fips defined: False
+    """.replace('    ', ''))
+
+def check(condition, expected):
+    """ Check condition against expected """
+    adjusted = relevancy_to_adjust(f"{condition}: False")[0]['when']
+    assert adjusted == expected
+
+
+# Valid rules
+
+def test_empty():
+    """ Empty relevancy """
+    assert relevancy_to_adjust('') == list()
+
+def test_comments(full):
+    """ Extract comments """
+    assert full[0]['because'] == 'feature has been added in Fedora 33'
+    assert full[1]['because'] == 'using logical operators'
+    assert full[2]['because'] == 'modify environment'
+
+def test_disable(mini, full):
+    """ Disable test """
+    assert mini[0]['enabled'] == False
+    assert full[0]['enabled'] == False
+    assert full[1]['enabled'] == False
+
+def test_environment(full):
+    """ Modify environment """
+    assert full[2]['environment'] == {'PHASES': 'novalgrind'}
+
+def test_condition(mini, full):
+    """ Expressions conversion """
+    assert mini[0]['when'] == 'distro ~= fedora'
+    assert full[0]['when'] == 'distro ~< fedora-33'
+    assert full[1]['when'] == 'component == firefox and arch == ppc64'
+    assert full[2]['when'] == 'arch == s390x'
+    assert full[3]['when'] == 'collection == httpd24 and fips is defined'
+
+def test_operators_basic():
+    """ Basic operators unchanged """
+    check('component = python', 'component == python')
+    check('component == python', 'component == python')
+    check('arch == s390x', 'arch == s390x')
+    check('arch != s390x', 'arch != s390x')
+
+def test_operators_distro():
+    """ Special handling for distro """
+    check('distro = fedora', 'distro ~= fedora')
+    check('distro == fedora', 'distro ~= fedora')
+    check('distro != fedora', 'distro ~!= fedora')
+    check('distro < fedora-33', 'distro ~< fedora-33')
+    check('distro > fedora-33', 'distro ~> fedora-33')
+    check('distro <= fedora-33', 'distro ~<= fedora-33')
+    check('distro >= fedora-33', 'distro ~>= fedora-33')
+
+def test_operators_product():
+    """ Special handling for product """
+    check('product = fedora', 'product ~= fedora')
+    check('product == fedora', 'product ~= fedora')
+    check('product != fedora', 'product ~!= fedora')
+    check('product < fedora-33', 'product ~< fedora-33')
+    check('product > fedora-33', 'product ~> fedora-33')
+    check('product <= fedora-33', 'product ~<= fedora-33')
+    check('product >= fedora-33', 'product ~>= fedora-33')
+
+def test_operators_special():
+    """ Check 'defined' and 'contains' """
+    check('fips defined', 'fips is defined')
+    check('fips !defined', 'fips is not defined')
+    check('collection contains http24', 'collection == http24')
+    check('collection !contains http24', 'collection != http24')
+
+
```

Comment:
Relevancy attribute was defined as `Should be a list of rules or a string with one rule per line.`
If I do list of rules, e.g
```
relevancy:
 - "distro > rhel-8: False"
 - "distro == rhel-7: Foo=bar"
```
lint tracebacks
```
Traceback (most recent call last):
  File "/home/lzachar/.virtualenvs/tmt/bin/tmt", line 7, in <module>
    exec(compile(f.read(), __file__, 'exec'))
  File "/home/lzachar/gits/tmt/bin/tmt", line 8, in <module>
    tmt.cli.main()
  File "/home/lzachar/.virtualenvs/tmt/lib/python3.9/site-packages/click/core.py", line 829, in __call__
    return self.main(*args, **kwargs)
  File "/home/lzachar/.virtualenvs/tmt/lib/python3.9/site-packages/click/core.py", line 782, in main
    rv = self.invoke(ctx)
  File "/home/lzachar/.virtualenvs/tmt/lib/python3.9/site-packages/click/core.py", line 1259, in invoke
    return _process_result(sub_ctx.command.invoke(sub_ctx))
  File "/home/lzachar/.virtualenvs/tmt/lib/python3.9/site-packages/click/core.py", line 1259, in invoke
    return _process_result(sub_ctx.command.invoke(sub_ctx))
  File "/home/lzachar/.virtualenvs/tmt/lib/python3.9/site-packages/click/core.py", line 1066, in invoke
    return ctx.invoke(self.callback, **ctx.params)
  File "/home/lzachar/.virtualenvs/tmt/lib/python3.9/site-packages/click/core.py", line 610, in invoke
    return callback(*args, **kwargs)
  File "/home/lzachar/.virtualenvs/tmt/lib/python3.9/site-packages/click/decorators.py", line 21, in new_func
    return f(get_current_context(), *args, **kwargs)
  File "/home/lzachar/gits/tmt/tmt/cli.py", line 334, in lint
    if not test.lint():
  File "/home/lzachar/gits/tmt/tmt/base.py", line 328, in lint
    metadata['adjust'] = tmt.convert.relevancy_to_adjust(relevancy)
  File "/home/lzachar/gits/tmt/tmt/convert.py", line 627, in relevancy_to_adjust
    for line in re.split(r'\s*\n\s*', relevancy.strip()):
AttributeError: 'list' object has no attribute 'strip'
```

Reply:
Another test proposal main.fmf and aaa.fmf, relevancy attribute was redefined.
main.fmf: ```relevancy: "distro > rhel-8: False"```
aaa.fmf: ```relevancy: "product == rhscl: False"```

After tmt lint onlfy `aaa.fmf` is changed. main.fmf keeps relevancy attrbute and thus /aaa test has bot relevancy AND adjust defined and any further `tmt  test lint `doesn't seem to change that.

Reply:
Hopefully last one, using relavancy+:
main.fmf: ```relevancy: "distro > rhel-8: False"```
aaa.fmf: ```relevancy+: "\nproduct == rhscl: False"```
After `tmt test lint -f` nothing is changed.

Reply:
The main motivation for the `tmt test lint --fix` functionality was to help those who have already mass-migrated their tests to git. Do you expect there will be many cases as this? Could we possibly handle these special cases later? I suggest to fix the list format now and address the (rather complex) inheritance scenarios in a separate pull request later.

Reply:
I hope not but I tried to use `relevancy+` in few cases. Imho it is OK to address these perks later.

Reply:
Good, thanks.

Reply:
The problem with list relevancy fixed and covered by tests.